### ISSUE TITLE
Possible differences between this version and official TensorFlow version

### ISSUE THREAD

**Issue by saw79** [rwightman/efficientdet-pytorch#92]
I'm working on a project that would like to use EfficientDet for an object detection problem. We need a PyTorch implementation and not a TensorFlow implementation, so this repo looked quite promising. I am able to verify that both versions perform almost identically on the COCO validation set (as advertised in the README). However, I'm getting very different results on my own toy dataset in my specific problem domain (this is all on pretrained/COCO weights). With the TensorFlow implementation, I get AP50 values ranging from 0.5 to 0.8 as I scale the architecture from D0 to D7x, but with this version my AP50's only go from 0.3 to 0.5 or so.

I initially discovered that the image_size configuration was broken (should probably make a separate issue for this when I get a chance) - at least when using the validate.py interface, so this version was using the default network size which is smaller than my images (1600x1200). After hardcoding the image size in the effdet/config file (to 1664 since it needs to be a multiple of stride), my performance improves by about 0.1 or so, but there's still a significant gap between this and the official TF version.

If anyone has any ideas on where else I could explore to see what differences there could be between versions, that would be much appreciated. Thanks!

**Comment by rwightman** [comment]
@saw79 I've been slowing working on improvements that haven't been merged to maste yetr, you may want to try the more_datasets branch https://github.com/rwightman/efficientdet-pytorch/tree/more_datasets and see how it goes

There are quite a few changes there aside from the obvious ones like support for VOC and OpenImages.

* a new loss function that closer matches newer changes to the TF impl that were made after my port, including label smoothing (`--legacy-focal`l to use old one as it's a bit better memory/speed wise)
* anchor labeler can operate in the collate (loader processes) to generate labels off of the critical path (`--loader-labeler` to try, but I may flip it around and make it the default)
* updated the anchor code to allow creating non-square anchor layouts, so for your case you could make a model config with 1664x1280 anchor layout
* proper support for fine-tuning from pretrained coco weights and change the head class (num_classes) or box (anchors), the head weights will be re-initialized properly
* support for native (non Apex AMP)

and probably more that I'm not remembering...

**Comment by rwightman** [comment]
Oh yeah, and loss fn can bit run in jit script or normal mode... there is a noteworthy memory vs speed tradeoff here so I created a `--no-jit-loss` flag to disable it as the jit'd loss usually runs faster but eats up more GPU memory (haven't figured that one out) ... I've noted some stability issues with jit loss + AMP (mixed-prec) in some cases, but forced a few values to have made it stable

**Comment by saw79** [comment]
Interesting, thanks for the feedback, I'll give that branch a try!
